### DISCUSSION TITLE
webhook condition - trigger on change

### DISCUSSION THREAD

**Discussion by jnc27** [netbox-community/netbox#13159] [Q&A]
I would like to fire the webhook if there was a change in a specific custom field.
I couldn't find a way to retrieve the values. I was thinking it could work like this:

        {
            "attr": "snapshots.prechange.custom_fields.cf_test",
            "negate": true,
            "value": "data.custom_fields.cf_test"
        }

**Comment by candlerb** [comment]
I think you've reached the limits of what's possible in webhook filtering. Your example looks for the case where the target field is not the literal value `"data.custom_fields.cf_test"`

Doing anything more sophisticated would involve changing Netbox, either to allow you to plug in python code here (as is possible with [custom validation](https://docs.netbox.dev/en/stable/customization/custom-validation/#custom-validation-logic)), or to use a more powerful expression language like [Jsonnet](https://brian-candler.medium.com/streamlining-kubernetes-application-deployment-with-jsonnet-711e15e9c665):

```
# A webhook condition expressed as a Jsonnet function
function(obj)
  obj.snapshots.prechange.custom_fields.cf_test != obj.snapshots.postchange.custom_fields.cf_test
```

```
local f = import 'cond.libsonnet';
local data = {
  snapshots: {
    prechange: {
      custom_fields: {
        cf_test: "abc",
      },
    },
    postchange: {
      custom_fields: {
        cf_test: "def",
      },
    },
  },
};
f(data)
```

But for now, I think this logic needs to live in your webhook receiver.

**Comment by 4ydan** [comment]
Maybe today its more feasable? My receiver is gitlab and I have no way there to compare the changes, I would need to write some in-between receiver to forward my request to gitlab.
